stdlib/format.mli

```diff
@@ -1246,6 +1246,21 @@ val pp_print_text : formatter -> string -> unit
   @since 4.02
 *)
 
+val format_text: ('a,'b,'c,'d,'e,'f) format6 -> ('a,'b,'c,'d,'e,'f) format6
+(** [text fmt] replaces spaces and newlines in the format string literal [fmt]
+    with hint breaks or forced newlines:
+- Only ascii spaces ([" " = "\u{20}"]) and linefeeds ["\n" = "\u{A}"] are
+  replaced. In particular, this means that non-breaking spaces (for instance
+  ["\u{A0}"]) are left unchanged.
+- [0<=k<=1] newlines followed by [n] spaces are converted to a [@;<0 (k+n)>]
+  break hint (when [k+n>=1]).
```

Comment:
Isn't that rule redundant with the space and newline replacing rules ? If not I'm not sure I understand its implications.

Reply:
Coalescing multiple spaces in a single break hint ensure that the line is split at most once in the beginning of the group of spaces. This ensures that the spaces in
```ocaml
format_text "a                                                                                                                                                          b"
```
is translated to a single line break rather than multiple line breaks between blank lines.

Reply:
Blank lines by themselves should map to a single `@\n`, it is the preceding line that must be a `\@n` too:

1. Blank lines  (lines made only of spaces) are replaced by `\@n`.
2. Sequences of spaces and a newline preceding a blank line are replaced by `\@n`: Blank lines remain blank lines in the output.
3. Remaining sequences made of `k` spaces and newlines are replaced by `@;<0 k>`.

Reply:
I have added the new description (with a tweak that the space width is at least 1 on break hints) and adapted the implementation to match the behavior and the specification.

Reply:
(Does @dbuenzli's thumb up mean that he approves the specification you converged towards? If so, please feel free to merge.)